src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2486,6 +2485,8 @@ private bool InstructionSelfDestruct<TTracing>(EvmState vmState, ref EvmStack<TT
             _codeCache.Set(codeHash, codeInfo);
         }
 
+        vmState.CreateList.Add(contractAddress);
```

Comment:
Minor observation: So I commented the 2488 line and I expected `self_destruct_in_initcode` to fail. However, the other tests failed:
![image](https://github.com/NethermindEth/nethermind/assets/9356351/d21d5ba3-ad08-4f48-90e6-c19124f7c720)

Reply:
I am a bit confused `InstructionCreate` (only modification) in this PR is changing behaviour of CREATE/CREATE2 but it doesn't affect initCode so I am curious if we still don't have bug

Reply:
You are correct with your observation. I should have made sure account is null, not codehash = Keccak.EmptyString.
Fixed